ingestion/src/airflow_provider_openmetadata/lineage/openmetadata.py

```diff
@@ -74,35 +126,118 @@ def get_lineage_config() -> OpenMetadataLineageConfig:
+    otherwise create it.
+
+    :param operator: task from which we extract the lienage
+    :param client: OpenMetadata API wrapper
+    :param config: lineage config
+    :return: PipelineService
+    """
+    operator.log.info("Get Airflow Service ID")
+    airflow_service_entity = client.get_by_name(
+        entity=PipelineService, fqdn=config.airflow_service_name
+    )
+
+    if airflow_service_entity is None:
+        pipeline_service = CreatePipelineServiceEntityRequest(
+            name=config.airflow_service_name,
+            serviceType=PipelineServiceType.Airflow,
+            pipelineUrl=conf.get("webserver", "base_url"),
+        )
+        airflow_service_entity = client.create_or_update(pipeline_service)
+        operator.log.info("Created airflow service entity {}", airflow_service_entity)
+
+    return airflow_service_entity
+
+
+def create_pipeline_entity(
+    dag_properties: Dict[str, str],
+    task_properties: Dict[str, str],
+    operator: "BaseOperator",
+    dag: "DAG",
+    airflow_service_entity: PipelineService,
+    client: OpenMetadata,
+) -> Pipeline:
+    """
+    Prepare the upsert the pipeline entity with the given task
+
+    :param dag_properties: attributes of the dag object
+    :param task_properties: attributes of the task object
+    :param operator: task being examined by lineage
+    :param dag: airflow dag
+    :param airflow_service_entity: PipelineService
+    :return: PipelineEntity
+    """
+    pipeline_service_url = conf.get("webserver", "base_url")
+    dag_url = f"{pipeline_service_url}/tree?dag_id={dag.dag_id}"
+    task_url = (
+        f"{pipeline_service_url}/taskinstance/list/"
+        + f"?flt1_dag_id_equals={dag.dag_id}&_flt_3_task_id={operator.task_id}"
+    )
+    dag_start_date = convert_epoch_to_iso(int(float(dag_properties["start_date"])))
+
+    downstream_tasks = []
+    if "_downstream_task_ids" in task_properties:
+        downstream_tasks = ast.literal_eval(task_properties["_downstream_task_ids"])
+
+    operator.log.info(f"downstream tasks {downstream_tasks}")
+
+    task_start_date = (
+        task_properties["start_date"].isoformat()
+        if "start_time" in task_properties
+        else None
+    )
+    task_end_date = (
+        task_properties["end_date"].isoformat()
+        if "end_time" in task_properties
+        else None
+    )
+
+    task = Task(
+        name=task_properties["task_id"],
+        displayName=task_properties.get("label"),  # v1.10.15 does not have label
+        taskUrl=task_url,
+        taskType=task_properties["_task_type"],
+        startDate=task_start_date,
+        endDate=task_end_date,
+        downstreamTasks=downstream_tasks,
+    )
+    create_pipeline = CreatePipelineEntityRequest(
+        name=dag.dag_id,
+        displayName=dag.dag_id,
+        description=dag.description,
+        pipelineUrl=dag_url,
+        startDate=dag_start_date,
+        tasks=[task],  # TODO: should we GET + append?
```

Comment:
Hi, @harshach! Here we are going to run a `PUT` on a Pipeline Entity named after the `DAG`. If I understood this correctly, the lineage is executed after each `task`. I believe that if we just set here `tasks=[task]`, we will end up with a Pipeline instance only keeping the last task being executed by the lineage.

Would it make sense to try to `GET` the pipeline, obtain its `tasks` and run a union with the newly processed `task` by the backend?

Thanks!

Reply:
@pmbrull Thanks a lot for this patch.
Yes you are right and this is a bug in recent refactor. However I was doing an update to the list of tasks previously on the server side.
I think what you are proposing makes sense but I am not sure if we should reconcile this on client side vs server.
If we do it on the server side, its hard to update the tasks that are deleted or renamed so we will keep adding to the existing list.
I guess doing it on the client side is better. One other option I was looking at is doing it via airflow rest apis to get all the workflows and get the pipeline created or create AirflowOpenMetadataOperator that users can attach to their dag from where we can get all the dag info at one go and publish to OM.
We will use lineage backend to read the inlets and outlets to extract the lineage only and not update the pipeline

Reply:
Another limitation with the current approach is getting workflow status of success/failed etc..
LineageBackend gets executed only when the dag/task successfully executed so any failing workflow we won't be able to capture the pipeline information at all.

Reply:
but it is also true that having the backend create already the pipelines makes it easier for developers as they don't need to add more boilerplate to the dags. We should then also properly define the expected order of execution, as before running the lineage there would be a dependency on another process ingesting the pipelines.

Maybe reconciling the tasks at client-side (to not mess too much with the backend and have total flexibility in `PUT`/`PATCH`), would be easier to set up. We should still need to find a proper way to extract the execution status of a pipeline though

Reply:
Looks like atlan is obtaining DAG & Task status. Might be worth a look https://github.com/atlanhq/atlan-lineage-airflow/blob/master/atlan/lineage/models/backend.py#L155

Reply:
Based on my testing the lineage backend will only be executed if the workflow is success, that means this status will always be "success". But I need to re-run that.
1. Continue to createOrUpdate from Lineage backend. 
2. Provide an option to directly integrate with rest apis to fetch all the workflows (airflow 2.0). Most of them are upgrading to 2.0 anyway.  Step 1 will still work as it only updates the workflow that are already created.
3. Provide an PythonOperator that can be attached to the dag in the code to send us lineage details

@pmbrull what you think of this approach?

Reply:
Agree complying on 1 and 2. I'll also take another look to the backend and see if we can obtain the status directly. It runs in an internal '_post_run' or something like that, so luckily that would always be executed.

Re 3 it might require further discussion, as the tricky part is at which point in the pipeline we execute the usual QA task without relying on upstream tasks not failing etc. if we manage to control it with the lineage backend we could skip that